x-pack/plugins/ml/server/saved_objects/checks.ts

```diff
@@ -154,5 +156,106 @@ export function checksFactory(
     };
   }
 
-  return { checkStatus };
+  async function deleteJobsCheck(
+    request: KibanaRequest,
+    jobType: JobType,
+    jobIds: string[],
+    spacesEnabled: boolean,
+    resolveMlCapabilities: ResolveMlCapabilities
+  ) {
+    if (jobType !== 'anomaly-detector' && jobType !== 'data-frame-analytics') {
+      throw Boom.badRequest('Job type must be "anomaly-detector" or "data-frame-analytics"');
+    }
+
+    const mlCapabilities = await resolveMlCapabilities(request);
+    if (mlCapabilities === null) {
+      throw Boom.internal('mlCapabilities is not defined');
+    }
+
+    if (
+      mlCapabilities.canDeleteJob === false ||
+      mlCapabilities.canDeleteDataFrameAnalytics === false
+    ) {
+      // user does not have access to delete jobs.
+      return jobIds.reduce((results, jobId) => {
+        results[jobId] = {
+          canDelete: false,
+          canUnTag: false,
+        };
+        return results;
+      }, {} as DeleteJobCheckResponse);
+    }
+
+    if (spacesEnabled === false) {
+      // spaces are disabled, delete only no untagging
+      return jobIds.reduce((results, jobId) => {
+        results[jobId] = {
+          canDelete: true,
+          canUnTag: false,
+        };
+        return results;
+      }, {} as DeleteJobCheckResponse);
+    }
+    const canCreateGlobalJobs = await jobSavedObjectService.canCreateGlobalJobs(request);
+
+    const jobObjects = await Promise.all(
+      jobIds.map((id) => jobSavedObjectService.getJobObject(jobType, id))
+    );
+
+    return jobIds.reduce((results, jobId) => {
+      const jobObject = jobObjects.find((j) => j?.attributes.job_id === jobId);
+      if (jobObject === undefined || jobObject.namespaces === undefined) {
+        // job saved object not found
+        results[jobId] = {
+          canDelete: false,
+          canUnTag: false,
+        };
+        return results;
+      }
+
+      const { namespaces } = jobObject;
+      const isGlobalJob = namespaces.includes('*');
+
+      // job is in * space, user can see all spaces - delete and no option to untag
+      if (canCreateGlobalJobs && isGlobalJob) {
+        results[jobId] = {
+          canDelete: true,
+          canUnTag: false,
+          // GLOBAL JOB BOOL????
+        };
+        return results;
+      }
+
+      // job is in * space, user cannot see all spaces - no untagging, no deleting
+      if (isGlobalJob) {
+        results[jobId] = {
+          canDelete: false,
+          canUnTag: false,
+        };
+        return results;
+      }
+
+      // jobs with are in individual spaces can only be untagged
+      // from current space if the job is in more than 1 space
+      const canUnTag = namespaces.length > 1;
```

Comment:
As above, should probably be `canUntag`.

Reply:
fixed https://github.com/elastic/kibana/pull/83871/commits/ceb41ead974a82cb78fa916b5466015f2f654b78